modules/node-shared/src/main/scala/io/constellationnetwork/node/shared/http/p2p/middlewares/FailureMiddleware.scala

```diff
@@ -0,0 +1,41 @@
+package io.constellationnetwork.node.shared.http.p2p.middlewares
+
+import cats.data.{Kleisli, OptionT}
+import cats.effect.Async
+import cats.effect.kernel.Clock
+import cats.implicits._
+
+import scala.concurrent.duration._
+
+import org.http4s.HttpRoutes
+
+object FailureMiddleware {
+
+  def failureSimulatorFeatureCheck[F[_]: Async](envVarName: String): F[Unit] =
+    Async[F].delay {
+      sys.env.get(envVarName).map(_.toLong)
+    }.flatMap {
+      case Some(failTime) =>
+        // Monotonic is off by about 30 seconds here, ideal to switch to in future.
+        // Clock[F].monotonic.flatMap { monotonic =>
+        Clock[F].realTime.flatMap { currentTime =>
```

Comment:
I needed something I could map to event log timings easily. The compose runner set a delay of like +90 seconds (before docker started,) and this was showing ~120-130 seconds at the same trigger time.

Reply:
I wonder if it has anything to do with running in a container environment? Could that additional overhead somehow mess with the assumptions of the cats scheduler?